From 10234378769bc3ce3d9b2f66f69b01870df5da1b Mon Sep 17 00:00:00 2001
From: "main()" <main@ehvag.de>
Date: Wed, 23 Nov 2016 02:13:20 +0100
Subject: [PATCH] Fix jmesmon#27

---
 src/daemon.rs | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/daemon.rs b/src/daemon.rs
index 7ec556d..2d3bedd 100644
--- a/src/daemon.rs
+++ b/src/daemon.rs
@@ -187,8 +187,8 @@ fn state_to_c_string(state: collections::HashMap<&str, &str>) -> ::std::ffi::CSt
 /// keys are defined as `STATE_*` constants in this module. Returns `true` if
 /// systemd was contacted successfully.
 pub fn notify(unset_environment: bool, state: collections::HashMap<&str, &str>) -> Result<bool> {
-    let c_state = state_to_c_string(state).as_ptr() as *const c_char;
-    let result = sd_try!(ffi::sd_notify(unset_environment as c_int, c_state));
+    let c_state = state_to_c_string(state);
+    let result = sd_try!(ffi::sd_notify(unset_environment as c_int, c_state.as_ptr()));
     Ok(result != 0)
 }
 
@@ -198,8 +198,8 @@ pub fn pid_notify(pid: pid_t,
                   unset_environment: bool,
                   state: collections::HashMap<&str, &str>)
                   -> Result<bool> {
-    let c_state = state_to_c_string(state).as_ptr() as *const c_char;
-    let result = sd_try!(ffi::sd_pid_notify(pid, unset_environment as c_int, c_state));
+    let c_state = state_to_c_string(state);
+    let result = sd_try!(ffi::sd_pid_notify(pid, unset_environment as c_int, c_state.as_ptr()));
     Ok(result != 0)
 }
 
