From 297953b3ffce2ec06fa73e28f0219740aa2cb665 Mon Sep 17 00:00:00 2001
From: Mike7R <mice7r@gmail.com>
Date: Fri, 29 Jul 2016 10:57:57 +0000
Subject: [PATCH] Support servers with SNI extension

---
 src/htsback.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/htsback.c b/src/htsback.c
index 671d8b5..e1c900d 100644
--- a/src/htsback.c
+++ b/src/htsback.c
@@ -2536,6 +2536,9 @@ void back_wait(struct_back * sback, httrackp * opt, cache_back * cache,
               // new session
               back[i].r.ssl_con = SSL_new(openssl_ctx);
               if (back[i].r.ssl_con) {
+                const char* hostname = jump_protocol_const(back[i].url_adr);
+                // some servers expect the hostname on the clienthello (SNI TLS extension)
+                SSL_set_tlsext_host_name(back[i].r.ssl_con, hostname);
                 SSL_clear(back[i].r.ssl_con);
                 if (SSL_set_fd(back[i].r.ssl_con, (int) back[i].r.soc) == 1) {
                   SSL_set_connect_state(back[i].r.ssl_con);
